File: DSRemote/Models/GameRom.swift
import Foundation

struct GameRom: Identifiable, Codable {
    let id = UUID()
    let name: String
    let fullPath: String
    let sizeFormatted: String

    enum CodingKeys: String, CodingKey {
        case name, fullPath, sizeFormatted
    }
}

File: DSRemote/Models/GameRom.cs
namespace DSRemote.Models;

public class GameRom
{
    public string Name { get; set; } = string.Empty;
    public string FullPath { get; set; } = string.Empty;
    public long SizeBytes { get; set; }
    public string SizeFormatted => FormatSize(SizeBytes);

    private static string FormatSize(long bytes)
    {
        string[] suffixes = { "B", "KB", "MB", "GB" };
        int order = 0;
        double size = bytes;
        while (size >= 1024 && order < suffixes.Length - 1)
        {
            order++;
            size /= 1024;
        }
        return $"{size:0.##} {suffixes[order]}";
    }
}
